source/base/parameter_handler.cc

```diff
@@ -1602,8 +1602,8 @@ ParameterHandler::recursively_print_parameters(
               }
 
             // print the name and (if set) value of this entry
-            out << std::setw(overall_indent_level * 2) << "set "
-                << demangle(p.first)
+            out << std::setw(overall_indent_level * 2) << ""
+                << "set " << demangle(p.first)
```

Comment:
The important point here is that the width set with `setw()` is used to put the following object into a box this many characters wide. If you then pass `""`, this creates a number of spaces. If you pass anything else, it gets put into that box. Here (as in the code before #17616), we really just want to pad with a few spaces, so we need to pass an empty string.

Reply:
Interesting case. Is there a reason we can't use `std::right` and `std::setfill(' ')` here? That would be idiomatic.

We should also add a test since we already broke this once.

Reply:
But I don't actually want to right-align anything. I just want to print a specific number of spaces, and I think the code expresses that. How else would you want to write it?

I'll think about a test.

Reply:
Yeah, what I suggested wouldn't work. I don't feel that strongly about it but we could accomplish the same thing via
```cpp
out << std::string(overall_indent_level * 2, ' ')
```
instead of using stream manipulators.

Reply:
I don't think that's much easier to read. I'll just add a comment.